### ISSUE TITLE
Update logic for cloudconfig in S3 bucket

### ISSUE THREAD

**Issue by rossf7** [giantswarm/aws-operator#533]
Towards giantswarm/giantswarm#2023

Currently we update the cloud config in the S3 bucket when clusters are reprocessed. This can cause problems if breaking changes are deployed.

- In `k8scloudconfig` we want to use semver to identify breaking / non breaking changes.
- We also need to version the cloudconfig stored in the S3 bucket.
- For non breaking changes we should update the existing cloud config and roll the nodes to apply the changes.
- For breaking changes a new version should be created in the S3 bucket and the cluster should not be updated.

## Cloud Config operator 

- Until the cloud config operator is in place `aws-operator` will need to determine which k8scloudconfig version to use.
- Once the cloud config operator is in place it will generate the cloud config but `aws-operator` will still need to upload it to S3.

**Comment by xh3b4sd** [comment]
@rossf7 please have a look again and sort the issue from an organizational point of view. <3

**Comment by rossf7** [comment]
Moving this to Team Batman as its part of the AWS updates story.

**Comment by rossf7** [comment]
Closing. This is ancient and superseded by work done on the updates story.
